.github/workflows/lint.yml

```diff
@@ -91,6 +91,13 @@ jobs:
         if: always()
         run: |
           (! git --no-pager grep -In '#!.*python$' -- . || (echo "The above lines have versionless Python shebangs; please specify either python2 or python3"; false))
+      - name: Ensure no trailing whitespaces
+        if: always()
+        run: |
+         # Since git apply will fix all the flags raised by the whitespace
+         # errors and this will not raise any warinig or error
+         git diff > patchname.patch
+         git apply --whitespace=fix patchname.patch    
```

Comment:
Since this is run on PRs, wouldn't this code simply modify the code in the run but not commit it anywhere? 

In general, we want our lint jobs to display a clear message to our users about how to fix the lint themselves so they could have full control of their PR.

Reply:
Hey @janeyx99 after continuously working on this issue for three days , i still dont know why my PR is failing the lint checks.
. I will be glad if you can help me in any way, btw i have made a pre commit hook that will check for trailing whitespace and then used that hook in the lint.yml file if a trailing whitespace is found it will raise a warning.
And one more thing , right now i just dont have any clue left if this solution does not work i am just confused how to proceed furtther. So please help me figure out the correct solution